src/xdp/trn_transit_scaled_endpoint.c

```diff
@@ -393,7 +393,7 @@ static __inline int trn_scaled_ep_decide(struct transit_packet *pkt)
 		pkt->inner_ipv4_tuple.sport;
 
 	pkt->scaled_ep_opt->scaled_ep_data.target.dport =
-		pkt->inner_ipv4_tuple.dport;
+		bpf_htons(ep->remote_ports[remote_idx]);
```

Comment:
minor: You can save this conversion in the data-path by calling ```htons``` in the rpc implementation.

Reply:
That makes sense, thanks for the catch!